### PR TITLE
Prepare context menu for main widget tabs

### PR DESCRIPTION
**Pull request by markus-moser**
image

### CODE REVIEW

File: assets/js/src/core/modules/widget-manager/hooks/use-context-menu.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      setContextMenuState({ x: event.clientX, y: event.clientY, tabNode: node })
+      node.getExtraData()
+    }
+  }
+
+  const closeContextMenu = (): void => {
+    setContextMenuState(null)
+  }
+
+  const handleClickOutside = (event: MouseEvent): void => {
```
Comment (vin0401) [inline assets/js/src/core/modules/widget-manager/hooks/use-context-menu.tsx]: @Corepex defined a hook for this, some time ago:
https://github.com/pimcore/studio-ui-bundle/blob/1.x/assets/js/src/core/utils/hooks/use-click-outside.ts

Would it make sense to use it here?

**Review by vin0401 — CHANGES_REQUESTED** [review]
Nice. :) Just one small thing. :)

File: assets/js/src/core/modules/widget-manager/hooks/use-context-menu.tsx (hunk 1 above)
Comment (markus-moser) [reply assets/js/src/core/modules/widget-manager/hooks/use-context-menu.tsx]: Ahh, cool thanks - looks like if it is exactly how we need it

Comment (markus-moser) [reply assets/js/src/core/modules/widget-manager/hooks/use-context-menu.tsx]: done

**Review by vin0401 — APPROVED** [review]

**Pull request merged**